### ISSUE TITLE
How to run add explicit imports without an editor

### ISSUE THREAD

**Issue by locallycompact** [haskell/haskell-language-server#3223]
Hi. I know there is an editor command you can invoke to add explicit import statements, but how do you just run the command to add explicit import statements from the command line across a whole project without an editor? I don't know if this is explained anywhere.

**Comment by michaelpj** [comment]
We don't generally export any HLS features via the CLI. That would be a pretty substantial feature request. (And also something that should be doable generically for any language server?)

**Comment by locallycompact** [comment]
Well I would not want to invoke this via an http interface, just a library call that runs add explicit imports on all the files.

**Comment by kokobd** [comment]
> (And also something that should be doable generically for any language server?)

I think the best way to do this is to write an LSP client as a CLI program? After a quick google search, the closest existing solution is: https://github.com/ii14/lsp-command I guess there is someway to operate vim non-interactively, so maybe this is a solution?

And then this is not a feature we are going to implement in HLS I think.

**Comment by michaelpj** [comment]
Yes, I think it's honestly very unlikely that this will ever happen unless someone who really cares decides to make a big project of it. I'll leave this open since it's a perfectly reasonable idea and it's good to record.
